diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,6 +122,168 @@ class LineLength : Rule("line-length") {
+                binaryText += astNode.treeParent.findChildByType(OPERATION_REFERENCE)!!.text
+                binaryText += astNode.text
+                binaryText += astNode.nextSibling { it.elementType == WHITE_SPACE}?.text ?: ""
+                if (leftOffset + binaryText.length > configuration.lineLength) {
+                    val commonParent = astNode.parent({ it in binList[index - 1].parents() })!!
+                    val nextNode = commonParent.findChildByType(OPERATION_REFERENCE)!!.treeNext
+                    if (!nextNode.text.contains("\n"))
+                        commonParent.appendNewlineMergingWhiteSpace(nextNode, nextNode)
+                    leftOffset = 0
+                    binaryText = astNode.text
+                    return
+                }
+            }
+        }
+    }
+
+    /**
+     * This method stored all the nodes that have BINARY_EXPRESSION or PREFIX_EXPRESSION element type.
+     * This method uses recursion to store binary node in the order in which they are located
+     * Also binList contains nodes with PREFIX_EXPRESSION element type ( !isFoo(), !isValid)
+     *@param node node in which to search
+     *@param binList mutable list of ASTNode to store nodes
+     */
+    private fun searchBinaryExpression(node: ASTNode, binList: MutableList<ASTNode>) {
+        if (node.hasChildOfType(BINARY_EXPRESSION) || node.hasChildOfType(PARENTHESIZED)) {
+            node.getChildren(null)
+                    .filter { it.elementType == BINARY_EXPRESSION || it.elementType == PARENTHESIZED }
+                    .forEach {
+                        searchBinaryExpression(it, binList)
+                    }
+        } else {
+            binList.add(node)
+            binList.add(node.treeParent.findChildByType(PREFIX_EXPRESSION) ?: return)
+        }
+    }
+
+    private fun dfsForProperty(node: ASTNode, binList: MutableList<ASTNode>) {
+        node.getChildren(null).forEach {
+            if (it.elementType in PROPERTY_LIST) {
+                if (it.elementType != REFERENCE_EXPRESSION || it.treeParent.elementType != SHORT_STRING_TEMPLATE_ENTRY)
+                    binList.add(it)
+            }
+            dfsForProperty(it, binList)
+        }
+    }
+
+    private fun fixProperty(parent: ASTNode, configuration: LineLengthConfiguration) {
+        if (!parent.hasChildOfType(STRING_TEMPLATE)) {
+            if (parent.hasChildOfType(BINARY_EXPRESSION)) {
+                val leftOffset = positionByOffset(parent.findChildByType(BINARY_EXPRESSION)!!.startOffset).second
+                fixCondition(parent, configuration, leftOffset, true)
+            }
+            if (parent.hasChildOfType(PARENTHESIZED)) {
+                val newParent = parent.findChildByType(PARENTHESIZED)!!
+                val leftOffset = positionByOffset(newParent.findChildByType(BINARY_EXPRESSION)!!.startOffset).second
+                fixCondition(newParent, configuration, leftOffset)
+            }
```

Comment:
```suggestion
                val newParent = parent.findChildByType(PARENTHESIZED) ?: parent
                val leftOffset = positionByOffset(newParent.findChildByType(BINARY_EXPRESSION)!!.startOffset).second
                fixCondition(newParent, configuration, leftOffset, !parent.hasChildOfType(PARENTHESIZED))
```
If these conditions cannot happen at the same time (can they?), this code could be collapsed into single branch.